### DISCUSSION TITLE
Unable to install on Ubuntu

### DISCUSSION THREAD

**Discussion by jpage4500** [shannah/jdeploy#149] [Q&A]
I only have a macbook M1 but installed VMWare Fusion and am running `Ubuntu 64-bit Arm Server 24.04` in a VM.

Anyway, I wasn't able to install my app via the command line.. sorry I can't copy/paste between the VM so a screenshot is the best I could get:
![image](https://github.com/user-attachments/assets/20da203a-3f64-4421-a085-17435817b62f)

The reason I was trying was I got a report from someone who also couldn't download and install the app directly. I pasted the commands he ran below:
```
127 user@ubu3523 ~/Downloads % sudo gdebi android-device-manager_24.10.21-1514-1729617452_amd64.deb
[sudo] password for user: 
/usr/bin/gdebi:113: SyntaxWarning: invalid escape sequence '\S'
  c = findall("[[(](\S+)/\S+[])]", msg)[0].lower()
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Reading state information... Done

Java desktop app to manage multiple Android devices via adb
Do you want to install the software package? [y/N]:y
/usr/bin/gdebi:113: FutureWarning: Possible nested set at position 1
  c = findall("[[(](\S+)/\S+[])]", msg)[0].lower()
dpkg: error processing archive android-device-manager_24.10.21-1514-1729617452_amd64.deb (--install):
 parsing file '/var/lib/dpkg/tmp.ci/control' near line 1:
 invalid package name in 'Package' field: character ' ' not allowed (only letters, digits and characters '-+._')
Errors were encountered while processing:
 android-device-manager_24.10.21-1514-1729617452_amd64.deb
```

In his case it's complaining about spaces in the package name.. maybe that's something I'm doing wrong?

Here's my [package.json](https://github.com/jpage4500/AndroidDeviceManager/blob/develop/package.json#L26) file.. I'm not using spaces AFAIK
```
    "name": "android-device-manager",
```

**Comment by shannah** [comment]
Is that just with the .deb release or also the installer?

**Reply by jpage4500** [reply]
In my case I tried the command-line script. Someone else reported to me not being able to install the .deb release. So, 2 different things in this one.

He did get the .bin to install though FWIW

**Reply by shannah** [reply]
The .deb issue is fixed now. Download the .deb installer again and it will work.  Tested on Ubuntu 24.04

**Reply by shannah** [reply]
The cli install issue is fixed now as well.

**Comment by shannah** [comment]
Tried the command-line installer in Ubuntu 20.04 running under WSL and it seemed to work fine - other than the HiDPI issue that you raised in another thread.

Perhaps related to Ubuntu 24.  Will set up a VM and test it out.

**Comment by shannah** [comment]
Both issues fixed now.  One issue with deb installer related to the generated package name containing illegal characters.  This was corrected in the installer to sanitize the package name.  The other issue was with the bash installer script that is downloaded from jdeploy.com, didn't work on some versions of linux.  Changed from using cURL to using wget in the problem cases.
